### PR TITLE
Video playback speed control

### LINKED ISSUE #116: Custom Video Playback Speed Control
### Checklist

- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Gallery/issues) or [closed](https://github.com/FossifyOrg/Gallery/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Gallery/discussions) or [closed](https://github.com/FossifyOrg/Gallery/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Gallery/blob/master/CONTRIBUTING.md).
- [ ] I optionally [donated](https://fossify.org/donate) to support the Fossify mission.

### Feature description

allows users to control the playback speed of videos (e.g., 0.75x, 1x, 1.25x, etc.).

### Why do you want this feature?

Increased accessibility.

### Additional information

Implementation ideas:
- **Granular speed control:** Adjust the speed with a slider or input box, offering increments of 0.1x or 0.25x.
- **Remember last used speed:** Optionally, remember the last chosen speed.
Thank you ♡

### PR DESCRIPTION
**Pull request by Natolu24**
#### What is it?
- [ ] Bugfix
- [x] Feature
- [ ] Codebase improvement

#### Description of the changes in your PR
- Add control to the video playback speed (longpress, then swip up/down to control)
   - It increments by 0.25x, with minimum speed of 0.25x and maximum of 5.0x
- Add setting to define the default playback speed
   - Possible values : x0.25 / x0.5 / x0.75 / x1.0 / x1.25 / x1.5 / x1.75 / x2.0 / x3.0 / x4.0 / x5.0
- Add visual indicator to the current playback speed, which appears when the playback speed is changed from the default value
   - And can be clicked to reset to the default speed value

#### Before/After Screenshots/Screen Record
- Before:
Screenshot_Gallery_debug_Before

- After:
Screenshot_Gallery_debug_After

#### Fixes the following issue(s)
- Fixes #116

#### Acknowledgement
- [x] I read the [contribution guidelines](https://github.com/FossifyOrg/Gallery/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Aga-C** [comment]
> Add control to the video playback speed (longpress, then swip up/down to control) 

It's counterintuitive. How somebody will know that it's done this way? Also, there's no visual guidance - I don't know when I can start moving my finger to change the speed. I think it should rather be done similarly as it's in Fossify Music Player (and similar to numerous other players like NewPipe or VLC), where we have an icon/button to change the playback speed, and it shows a normal slider.

> It increments by 0.25x, with minimum speed of 0.25x and maximum of 5.0x

Why this range? I doubt anybody will need speed higher than 3x. In Fossify Music Player we have range between 0.25x to 3x without limiting to multiplies of 0.25. Same is in NewPipe.

> Add visual indicator to the current playback speed, which appears when the playback speed is changed from the default value

Why it displays on left if it's slower and on right if it's faster? It should stay in one place, moving elements are annoying for users.

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -181,6 +181,18 @@
...
+    <string name="default_playback_speed_2">x0.5</string>
+    <string name="default_playback_speed_3">x0.75</string>
+    <string name="default_playback_speed_4">x1.0</string>
+    <string name="default_playback_speed_5">x1.25</string>
+    <string name="default_playback_speed_6">x1.5</string>
+    <string name="default_playback_speed_7">x1.75</string>
+    <string name="default_playback_speed_8">x2.0</string>
+    <string name="default_playback_speed_9">x3.0</string>
+    <string name="default_playback_speed_10">x4.0</string>
+    <string name="default_playback_speed_11">x5.0</string>
```
Comment (Aga-C) [inline app/src/main/res/values/strings.xml:195]: 1. It shouldn't be in strings. Numeric value should be provided and formatted by the code according to current user region, not hard-coded. Check how it's done in Fossify Music Player.
2. It should be `0.25x`, `0.5x` etc.

**Comment by Natolu24** [comment]
Thanks for the review, it all does indeed makes sense, we'll fix all that !

**Comment by Natolu24** [comment]
Ok so we somewhat copy pasted the implementation from the Fossify Music Player to here :

Screenshot_20240515_201011_Gallery_debug(1)

Does the location look fine, or the UI itself need some adjustment ? (I guess adding a background to the icon could be nice)

Also, there's still a problem that we have issue finding a solution to :
When changing the playback speed, and then changing to a video that has already been started/opened, the playback speed won't be taken into account in the video nor the UI text.
I'm thinking that changing the playback speed should then affect all the videos currently started, but I do not really know how exactly it works, and how to implement it right, would someone have an idea / hint where to change things ?

**Comment by Aga-C** [comment]
- Background should be added, so it would look similarly to play button. Now the button is invisible on the light theme with horizontal videos.
image
- After checking in settings `Always open videos on a separate screen with new horizontal gestures`, there's no button for changing speed. Also, it always plays videos with 1x speed in this mode.
- When you change speed to lower than 1x, the icon changes, what's the same as in Music Player. But when I open a video, there's always an icon with a running man, no matter the speed. Icon should always be adjusted to the speed.

**Comment by Natolu24** [comment]
Background added, and fixed the rest !
Screenshot_20240516_161948_Gallery(1)

**Comment by Aga-C** [comment]
Seems to work fine, thanks! Now please wait for the final review by Naveen 🙂

**Comment by naveensingh** [comment]
Just made some XML layout changes. Not too happy with the playback speed button placement but I don't have any ideas at the moment either so I guess I'll merge this now and improve it later. Thanks!

**Pull request merged**